dnf/base.py

```diff
@@ -630,19 +630,18 @@ def _goal2transaction(self, goal):
             # Also keep RPMs with the same name - they're not always in all_obsoleted.
             obs = [i for i in obs if i in all_obsoleted or i.name == pkg.name]
 
-            # TODO: move to libdnf: getBestReason
-            reason = goal.get_reason(pkg)
+            reason = None
+            # inherit the best reason from obsoleted packages
             for obsolete in obs:
-                if reason == libdnf.transaction.TransactionItemReason_USER:
-                    # we already have a reason with highest priority
-                    break
-
                 reason_obsolete = ts.get_reason(obsolete)
-                if reason_obsolete == libdnf.transaction.TransactionItemReason_USER:
+                if reason is None:
                     reason = reason_obsolete
-                elif reason_obsolete == libdnf.transaction.TransactionItemReason_GROUP:
-                    if not self.history.group.is_removable_pkg(pkg.name):
-                        reason = reason_obsolete
+                elif libdnf.transaction.TransactionItemReasonCompare(reason, reason_obsolete) < 1:
```

Comment:
I understand you can't use the C++ operators, because the constants are `int`s with values that don't adhere to the order... but this is quite ugly and non-pythonic...

How about turning the function into something like:
```python
if libdnf.transaction.is_reason_stronger(reason, reason_obsolete):  # returns boolean
    reason = reason_obsolete
```
Which has the problem that it is not clear which should be the stronger reason, so maybe better:
```python
reason = libdnf.transaction.get_stronger_reason(reason, reason_obsolete)
```
Not ideal (and not that universal), but seems nicer to me.

Reply:
I do not consider it as a critical part of PR.